src/Databases/DataLake/DatabaseDataLake.cpp

```diff
@@ -0,0 +1,534 @@
+                {
+                    return std::make_shared<StorageLocalDeltaLakeConfiguration>();
+                }
+                /// Fake storage in case when catalog store not only
+                /// primary-type tables (DeltaLake or Iceberg), but for
+                /// examples something else like INFORMATION_SCHEMA.
+                /// Such tables are unreadable, but at least we can show
+                /// them in SHOW CREATE TABLE, as well we can show their
+                /// schema.
+                /// We use local as substitution for fake because it has 0
+                /// dependencies and the most lightweight
+                case DB::DatabaseDataLakeStorageType::Other:
+                {
+                    return std::make_shared<StorageLocalDeltaLakeConfiguration>();
+                }
+                default:
+                    throw Exception(ErrorCodes::BAD_ARGUMENTS,
+                                    "Server does not contain support for storage type {}",
+                                    type);
+            }
+        }
+    }
+}
+
+std::string DatabaseDataLake::getStorageEndpointForTable(const DataLake::TableMetadata & table_metadata) const
+{
+    auto endpoint_from_settings = settings[DatabaseDataLakeSetting::storage_endpoint].value;
+    if (endpoint_from_settings.empty())
+        return table_metadata.getLocation();
+    else
+        return table_metadata.getLocationWithEndpoint(endpoint_from_settings);
+
+}
+
+bool DatabaseDataLake::empty() const
+{
+    return getCatalog()->empty();
+}
+
+bool DatabaseDataLake::isTableExist(const String & name, ContextPtr /* context_ */) const
+{
+    const auto [namespace_name, table_name] = parseTableName(name);
+    return getCatalog()->existsTable(namespace_name, table_name);
+}
+
+StoragePtr DatabaseDataLake::tryGetTable(const String & name, ContextPtr context_) const
+{
+    return tryGetTableImpl(name, context_, false);
+}
+
+StoragePtr DatabaseDataLake::tryGetTableImpl(const String & name, ContextPtr context_, bool lightweight) const
+{
+    auto catalog = getCatalog();
+    auto table_metadata = DataLake::TableMetadata().withSchema().withLocation();
+    if (lightweight)
+        table_metadata = table_metadata.withLightweight();
+
+    const bool with_vended_credentials = settings[DatabaseDataLakeSetting::vended_credentials].value;
+    if (with_vended_credentials)
+        table_metadata = table_metadata.withStorageCredentials();
+
+    table_metadata = table_metadata.withDataLakeSpecificMetadata();
+
+    auto [namespace_name, table_name] = parseTableName(name);
+
+    if (!catalog->tryGetTableMetadata(namespace_name, table_name, table_metadata))
+        return nullptr;
+
+    /// Take database engine definition AST as base.
+    ASTStorage * storage = database_engine_definition->as<ASTStorage>();
+    ASTs args = storage->engine->arguments->children;
+
+    if (table_metadata.hasLocation())
```

Comment:
It does not make sense to check for `hasLocation` if above we requested location to be present with `withLocation`. I understand it is here because of `withLightweight`, but I don't get why we need `withLightweight` at all - just don't call `withLocation`/etc if you don't need it - by default it should not fetch anything.